File: file-chooser-library/filechooserlibrary/src/main/java/com/randomlychosenbytes/filechooser/RetainableDialogFragment.java
package com.randomlychosenbytes.filechooser;

import android.app.Dialog;
import android.app.DialogFragment;

public class RetainableDialogFragment extends DialogFragment {

    public RetainableDialogFragment() {
        setRetainInstance(true);
    }

    @Override
    public void onDestroyView() {
        Dialog dialog = getDialog();
        // handles https://code.google.com/p/android/issues/detail?id=17423
        if (dialog != null && getRetainInstance()) {
            dialog.setDismissMessage(null);
        }
        super.onDestroyView();
    }
}

File: file-chooser-library/filechooserlibrary/src/main/kotlin/com/randomlychosenbytes/filechooser/RetainableDialogFragment.kt
package com.randomlychosenbytes.filechooser

import androidx.fragment.app.DialogFragment

open class RetainableDialogFragment : DialogFragment() {
    override fun onDestroyView() {
        val dialog = dialog
        // handles https://code.google.com/p/android/issues/detail?id=17423
        if (dialog != null && retainInstance) {
            dialog.setDismissMessage(null)
        }
        super.onDestroyView()
    }

    init {
        retainInstance = true
    }
}